undercrawler/crazy_form_submitter.py

```diff
@@ -72,3 +74,17 @@ def _fill_search_form(search_term, form, meta, do_random_refinement=False):
 def _is_refinement_input(input_type, input_el):
     return (input_type == 'search category / refinement' and
             getattr(input_el, 'type', None) in ['checkbox'])
+
+
+class SplashFormRequest(SplashRequest, FormRequest):
+    def __init__(self, url=None, callback=None, method=None, formdata=None,
+                 body=None, **kwargs):
+        # First init FormRequest to get url, body and method
+        if formdata:
+            FormRequest.__init__(
+                self, url=url, method=method, formdata=formdata)
+            url, method, body = self.url, self.method, self.body
+        # Then pass all other kwargs to SplashRequest
+        SplashRequest.__init__(
+            self, url=url, callback=callback, method=method, body=body,
```

Comment:
`FormRequest.__init__` calls `escape_ajax` in constructor (via `_set_request`); it means URLs with `#!` fragments won't be processed correctly by SplashRequest